src/sas/sascalc/invariant/invariant.py

```diff
@@ -810,23 +813,39 @@ def get_surface(self, contrast, porod_const, extrapolation=None):
         """
         Compute the specific surface from the data.
 
-        Implementation::
+        Historically, Sv was computed with the invariant and the Porod
+        constant so as not to have to know the contrast in order to get the
+        Sv as:
+            surface = (pi * V * (1- V) * porod_const) / q_star
+        However, that turns out to be a pointless exercise since it
+        also requires a knowledge of the volume fractions and from the
+        volumer fraction and the invariant the contrast can be calculated as:
+            contrast**2 = q_star / (2 * pi**2 * V * (1- V))
 
-          V =  self.get_volume_fraction(contrast, extrapolation)
+        Thus either way, mathematically it is always identical to computing
+        with only the contrast and the Porod Constant. up to and including
+        SasView versions 4.2.2 and 5.0.1 the implentation used the traditional
+        circular approach.
 
-          Compute the surface given by:
-            surface = (2*pi *V(1- V)*porod_const)/ q_star
+        Implementation: ::
+           Given the above, as of SasView 4.3 and 5.0.2 we compute Sv simply
+           from the Porod Constant and the contrast between the two phases as:
+             surface = porod_const / (2 * pi contrast**2) 
 
-        :param contrast: contrast value to compute the volume
-        :param porod_const: Porod constant to compute the surface
-        :param extrapolation: string to apply optional extrapolation
+
+        :param contrast: contrast between the two phases
+        :param porod_const: Porod constant
+        :param extrapolation: string to apply optional extrapolation. This will
+               only be needed if and when the contrast term is calculated from
+               the invariant.
 
         :return: specific surface
         """
-        # Compute the volume
-        volume = self.get_volume_fraction(contrast, extrapolation)
-        return 2 * math.pi * volume * (1 - volume) * \
-            float(porod_const) / self._qstar
+        # convert porod_const to units of A^-5 instead of cm^-1 A^-4 so that
+        # s is returned in units of 1/A.
+        _porod_const = 1.0e-8 * porod_const
+        return float(_porod_const) / (2 * math.pi\
+                                     * math.fabs(float(contrast))**2)
```

Comment:
No need for `\` to join lines when the expression is in parentheses.

Also, you don't need to convert from integer to float in python 3, or in python 2 if you have `from __future__ import division` at the top of your file, so:
```python
return _porod_const / (2 * math.pi * abs(contrast)**2)
```

Reply:
Thanks - was wondering why so many "float" statements were being used but not in other code. can I remove all "float(x)" statements in the file then?

Reply:
can this now be set as "resolved"

Reply:
float() will convert strings to double, so some of them may still be needed, depending on the code.

It will also convert numpy float32 to double, but that will happen with pretty much any arithmetic operation in python except if you are combining it with another float32.